shield.go

```diff
@@ -17,6 +18,8 @@ const MaxLength = 4096
 // SaltLength is the length of the salt generated by GenerateSalt.
 const SaltLength = 32
 
+var ErrPasswordTooLong = errors.New("password too long")
```

Comment:
Interesting. After reading the article I think that even though I don't love the idea of having to define the `Error` type, I think it makes more sense especially because we get to treat them as constants.